processor/schemaprocessor/DESIGN.md

```diff
@@ -0,0 +1,28 @@
+# Design
```

Comment:
Thank you for adding this document, it is very useful.

Reply:
It is really awesome, and I also had no idea github support mermaid diagrams. 

100% going to start doing this more often.

Reply:
Yay I'm glad ya'll like it!  I contributed on to the collector too at https://github.com/open-telemetry/opentelemetry-collector/blob/main/docs/internal-architecture.md.  I hope to make more!